erpnext/accounts/doctype/pricing_rule/utils.py

```diff
@@ -398,7 +398,9 @@ def get_qty_and_rate_for_other_item(doc, pr_doc, pricing_rules):
 				pricing_rules[0].apply_rule_on_other_items = items
 				return pricing_rules
 
-def get_qty_amount_data_for_cumulative(pr_doc, doc, items=[]):
+def get_qty_amount_data_for_cumulative(pr_doc, doc, items=None):
+	if not items:
```

Comment:
```suggestion
	if items is None:
```

In these cases being explicit is probably safer. There was a case where a refactor like this broke some function because `{}` was passed and the calling function expected a mutated input but the function created a new dictionary instead.  🤦

Reply:
^ also applies to all other changes in this PR.